pypeit/scripts/setup_coadd2d.py

```diff
@@ -146,31 +161,32 @@ def main(args):
         cfg = {} if args.clean_par or pypeitFile is None else dict(pypeitFile.config)
         if par is None:
             # This is only used to get the default offsets and weights (see below)
-            par = load_spectrograph(spec_name).default_pypeit_par()
-            utils.add_sub_dict(cfg, 'rdx')
-            cfg['rdx']['scidir'] = sci_dir.name
-        else:
-            utils.add_sub_dict(cfg, 'rdx')
-            cfg['rdx']['redux_path'] = par['rdx']['redux_path']
-            cfg['rdx']['scidir'] = par['rdx']['scidir']
-            cfg['rdx']['qadir'] = par['rdx']['qadir']
-            utils.add_sub_dict(cfg, 'calibrations')
-            cfg['calibrations']['calib_dir'] = par['calibrations']['calib_dir']
+            par = load_spectrograph(spec_name).config_specific_par(spec2d_files[0])
+        utils.add_sub_dict(cfg, 'rdx')
+        cfg['rdx']['redux_path'] = par['rdx']['redux_path']
+        cfg['rdx']['scidir'] = par['rdx']['scidir']
+        cfg['rdx']['qadir'] = par['rdx']['qadir']
+        utils.add_sub_dict(cfg, 'calibrations')
+        cfg['calibrations']['calib_dir'] = par['calibrations']['calib_dir']
         utils.add_sub_dict(cfg, 'coadd2d')
         cfg['coadd2d']['offsets'] = par['coadd2d']['offsets'] \
                 if args.offsets is None else args.offsets
         cfg['coadd2d']['weights'] = par['coadd2d']['weights'] \
                 if args.weights is None else args.weights
+        cfg['coadd2d']['spat_toler'] = par['coadd2d']['spat_toler'] \
+                if args.spat_toler is None else args.spat_toler
 
         # Build the default parameters
-        cfg = CoAdd2D.default_par(spec_name, inp_cfg=cfg, det=args.det, slits=args.only_slits)
+        cfg = CoAdd2D.default_par(spec_name, inp_cfg=cfg, det=args.det, only_slits=args.only_slits,
+                                  exclude_slits=args.exclude_slits)
 
         # Create a coadd2D file for each object
-        # NOTE: Below expect all spec2d files have the same path
         for obj, files in object_spec2d_files.items():
             tbl = Table()
             tbl['filename'] = [f.name for f in files]
-            ofile = args.pypeit_file.replace('.pypeit', f'_{obj}.coadd2d')
-            inputfiles.Coadd2DFile(config=cfg, file_paths=[str(sci_dir)],
+            ofile = \
+                str(Path(cfg['rdx']['redux_path']) / Path(args.pypeit_file).name.replace('.pypeit', f'_{obj}.coadd2d')) \
+                if args.pypeit_file is not None else str(Path(cfg['rdx']['redux_path']) / f'{spec_name}_{obj}.coadd2d')
```

Comment:
I'm partial to long single statements 😄 , but I would break this into two:
```
ofile = f'{spec_name}_{obj}.coadd2d' if args.pypeit_file is None \
            else Path(args.pypeit_file).name.replace('.pypeit', f'_{obj}.coadd2d'))
ofile = str(Path(cfg['rdx']['redux_path'] / ofile)
```

Reply:
Thanks! Done.